File: android/src/main/java/expo/modules/docusign/DocuSignError.kt
package expo.modules.docusign

import expo.modules.kotlin.exception.CodedException

class NotInitializedException :
  CodedException("DocuSign SDK has not been initialized. Call initialize() first.")

class NotLoggedInException :
  CodedException("DocuSign SDK is not logged in. Call loginWithAccessToken() first.")

class LoginFailedException(message: String) :
  CodedException("DocuSign login failed: $message")

class SigningFailedException(message: String) :
  CodedException("DocuSign signing failed: $message")

class PresentationException(message: String) :
  CodedException("Failed to present DocuSign signing UI: $message")

File: ios/DocuSignError.swift
import ExpoModulesCore

internal class NotInitializedException: Exception {
  override var reason: String {
    "DocuSign SDK has not been initialized. Call initialize() first."
  }
}

internal class NotLoggedInException: Exception {
  override var reason: String {
    "DocuSign SDK is not logged in. Call loginWithAccessToken() first."
  }
}

internal class PresentationException: GenericException<String> {
  override var reason: String {
    "Failed to present DocuSign signing UI: \(param)"
  }
}

internal class SigningFailedException: GenericException<String> {
  override var reason: String {
    "DocuSign signing failed: \(param)"
  }
}

internal class LoginFailedException: GenericException<String> {
  override var reason: String {
    "DocuSign login failed: \(param)"
  }
}
